### ISSUE TITLE
v1.15.0 fails to build due to missing CFLAGS

### ISSUE THREAD

**Issue by lachs0r** [netdata/netdata#6066]
##### Bug report summary
When attempting to build netdata 1.15.0, compilation fails because the compiler command lacks most of the CFLAGS from the .pc files that were found by configure on the system. The exact same system configuration worked with 1.14.0.

##### Build log excerpt
```
[   82s] + CFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g'
[   82s] + export CFLAGS
[   82s] + CXXFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g'
[   82s] + export CXXFLAGS
[   82s] + FFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g'
[   82s] + export FFLAGS
[   82s] + ./configure --host=x86_64-suse-linux-gnu --build=x86_64-suse-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/lib --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-dependency-tracking --docdir=/usr/share/doc/packages/netdata-1.15.0 --enable-plugin-nfacct --enable-plugin-freeipmi --enable-plugin-cups --with-zlib --with-math --with-user=netdata
```

```
[   91s] checking for libmnl... yes
[   91s] checking for mnl_socket_open in -lmnl... yes
[   91s] checking if nfacct.plugin should be enabled... yes
```

```
[   98s] gcc -DHAVE_CONFIG_H -I.   -DTARGET_OS=1 	-DVARLIB_DIR="\"/var/lib/netdata\"" 	-DCACHE_DIR="\"/var/cache/netdata\"" 	-DCONFIG_DIR="\"/etc/netdata\"" 	-DLIBCONFIG_DIR="\"/usr/lib64/netdata/conf.d\"" 	-DLOG_DIR="\"/var/log/netdata\"" 	-DPLUGINS_DIR="\"/usr/lib/netdata/plugins.d\"" 	-DRUN_DIR="\"/var/run/netdata\"" 	-DWEB_DIR="\"/usr/share/netdata/web\""     -I/usr/include/uuid -lcap     -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -pthread -flto -c -o collectors/nfacct.plugin/plugin_nfacct.o collectors/nfacct.plugin/plugin_nfacct.c
[   98s] collectors/nfacct.plugin/plugin_nfacct.c:17:10: fatal error: libmnl/libmnl.h: No such file or directory
[   98s]  #include <libmnl/libmnl.h>
[   98s]           ^~~~~~~~~~~~~~~~~
[   98s] compilation terminated.
[   98s] make[2]: *** [Makefile:2807: collectors/nfacct.plugin/plugin_nfacct.o] Error 1
[   98s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/netdata-v1.15.0'
[   98s] make[2]: *** Waiting for unfinished jobs....
[   98s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/netdata-v1.15.0'
```

```
$ pkg-config --cflags libmnl
-I/usr/include/libmnl

$ tree /usr/include/libmnl
/usr/include/libmnl
└── libmnl
    └── libmnl.h

1 directory, 1 file
```

##### OS / Environment
openSUSE Tumbleweed

**Comment by paulkatsoulakis** [comment]
Hello @lachs0r Could you provide more details on your build environment and what steps you followed exactly to end up on the error ?

**Comment by lachs0r** [comment]
The build environment is openSUSE’s build service. You can see the RPM spec file and full build logs here:

https://build.opensuse.org/package/show/network/netdata

**Comment by lachs0r** [comment]
I suspect this happens due to the CFLAGS/CXXFLAGS overrides. It appears the build system has changed a bit and now handles the flags incorrectly, so that the autodetected values will be replaced by environment variables (except for libuuid).

**Comment by cakrit** [comment]
We've marked 1.15.0 as a pre-release, until we resolve this. Thanks for the immediate report @lachs0r, I wish it had been caught earlier, but we'll get it fixed.

**Comment by mfundul** [comment]
@lachs0r I think it's relevant to update your dependencies to support the new functionality of the DB engine  as shown [here](https://github.com/netdata/netdata/blob/master/packaging/installer/README.md#prepare-your-system). The libraries required are libuv v1, liblz4, Judy and openssl.

**Comment by lachs0r** [comment]
> @lachs0r I think it's relevant to update your dependencies to support the new functionality of the DB engine as shown [here](https://github.com/netdata/netdata/blob/master/packaging/installer/README.md#prepare-your-system). The libraries required are libuv v1, liblz4, Judy and openssl.

Yes, I’m going to do that once it actually builds.

**Comment by paulkatsoulakis** [comment]
Hello @lachs0r, we nailed the root cause for your problems. 
It was an unfortunate style fix that went wrong and we are really sorry for letting this slip through.
Next steps from us:
1) Fix is merged already and we 're arranging to update v1.15.0 within the next couple of hours (Execution process takes time too).
2) I 'll be working on further additions of test cases on our CI, to block such bugs in the future

I hope this glitch won't stop you from continuing working with netdata in the future, thanks again for catching this. 
We 'll be here for you in the future if you face any other problems that we could help you with! 
See ya!
